src/client/nav/nav.js

```diff
@@ -786,9 +786,9 @@ spf.nav.handleNavigateSuccess_ = function(options, info, url, response) {
       var name = response['name'] || '';
       if (multipart) {
         var parts = response['parts'];
-        for (var i = 0; i < parts.length; i++) {
-          name = parts[i]['name'] || name;
-        }
+        spf.array.each(parts, function(part) {
```

Comment:
If I'm being really nitpicky, this would be faster by iterating in reverse and bailing on the first truthy value.

But that said, there are usually just a couple of parts, so it would likely never matter.